### PR TITLE
[linker] Adds more tests

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mcs/tools/linker/Makefile [hunk 1]
```diff
@@ -79,15 +82,23 @@ System/test-%.exe mscorlib/test-%.exe:
 	@rm -rf $(LINKER_OUTPUT)
 	@mkdir $(LINKER_OUTPUT)
 	@echo Testing $@
-	$(LINKER) -a Tests/$@
+	$(LINKER_DEFAULT) -a Tests/$@
+	$(TEST_EXEC) $(LINKER_OUTPUT)/$(@F)
+#	@rm -rf $(LINKER_OUTPUT)
```
Comment (akoeplinger) [inline mcs/tools/linker/Makefile]: uncomment this again.

File: mcs/tools/linker/Makefile [hunk 2]
```diff
@@ -70,7 +72,8 @@ endif
 
 LINKER_OUTPUT := illink-output-$(PROFILE_DIRECTORY)
 PROFILE_PATH = $(topdir)/class/lib/$(PROFILE_DIRECTORY)
-LINKER = MONO_PATH=$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE) $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/monolinker.exe -c link -out $(LINKER_OUTPUT) -b true -d $(PROFILE_PATH) $(LINKER_PROFILE_OPTIONS)
+LINKER = MONO_PATH=$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE) $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/monolinker.exe -c link  -out $(LINKER_OUTPUT) -b true -d $(PROFILE_PATH)
```
Comment (akoeplinger) [inline mcs/tools/linker/Makefile]: nit: double space before `  -out` 😄

File: mcs/tools/linker/Tests/mscorlib/test-calendar-02.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Linker options has to be: -l mideast
```
Comment (akoeplinger) [inline mcs/tools/linker/Tests/mscorlib/test-calendar-02.cs]: I assume this should be `-l none` ?

Comment (marek-safar) [reply mcs/tools/linker/Tests/mscorlib/test-calendar-02.cs]: yep

**Comment by marek-safar** [comment]
@monojenkins commit csproj

**Pull request merged**